### PR TITLE
support EH SCS binder batch sending

### LINKED ISSUE #25228: [FEATURE REQ] Enable batch-sending for spring cloud stream eventhubs binder
### Context
Both Spring Cloud Stream and Event Hubs Java SDK supports the feature of batch-sending.

Referring to [Spring Cloud Stream](https://docs.spring.io/spring-cloud-stream/docs/3.1.5/reference/html/spring-cloud-stream.html#_batch_producers), developers can use below API for sending a batch of messages:
   ```
   public Function<String, List<Message<String>>> batch();
   public Function<String, Message<String>> batch();
   public Supplier<Message<String>> batch();
   ```
Event Hubs also supports specifying the buffer size in bytes or message number for a batch.

### Problem
We intend to support batch-sending for Event Hubs binder. While currently we are blocked by the SDK batch-sending implementation, which doesn't provide a buffered producer that can batch and send messages automatically.

### Reference
[Kafka](https://cloud.spring.io/spring-cloud-stream-binder-kafka/spring-cloud-stream-binder-kafka.html#kafka-producer-properties) binder support: buffer size in bytes or buffer time out
[Rabbit](https://docs.spring.io/spring-cloud-stream-binder-rabbit/docs/current/reference/html/spring-cloud-stream-binder-rabbit.html#rabbit-prod-props) binder support: message number, buffer size in bytes or buffer time out.

### PR DESCRIPTION
**Pull request by yiliuTo**
Resolve #25228.

### CODE REVIEW

File: sdk/spring/spring-cloud-azure-stream-binder-eventhubs/CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,7 @@
 ## 4.0.0-beta.1 (Unreleased)
 
 ### Features Added
+- Support batch sending.
```
Comment (stliu) [inline sdk/spring/spring-cloud-azure-stream-binder-eventhubs/CHANGELOG.md:6]: shouldn't we add issue link here?d

File: sdk/spring/spring-messaging-azure-eventhubs/src/main/java/com/azure/spring/eventhubs/core/producer/BatchableProducerAsyncClient.java [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+    @Override
+    public Mono<Void> send(Flux<EventData> events) {
+        return send(events, null);
+    }
+
+    @Override
+    public Mono<Void> send(Flux<EventData> events, PartitionSupplier partitionSupplier) {
+        CreateBatchOptions options = buildCreateBatchOptions(partitionSupplier, maxBatchInBytes);
+
+        if (currentBatch == null) {
```
Comment (stliu) [inline sdk/spring/spring-messaging-azure-eventhubs/src/main/java/com/azure/spring/eventhubs/core/producer/BatchableProducerAsyncClient.java:46]: is this thread safe?

File: sdk/spring/spring-messaging-azure-eventhubs/src/main/java/com/azure/spring/eventhubs/core/producer/BatchableProducerAsyncClient.java [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+    }
+
+    @Override
+    public Mono<Void> send(Flux<EventData> events) {
+        return send(events, null);
+    }
+
+    @Override
+    public Mono<Void> send(Flux<EventData> events, PartitionSupplier partitionSupplier) {
+        CreateBatchOptions options = buildCreateBatchOptions(partitionSupplier, maxBatchInBytes);
```
Comment (stliu) [inline sdk/spring/spring-messaging-azure-eventhubs/src/main/java/com/azure/spring/eventhubs/core/producer/BatchableProducerAsyncClient.java:44]: why build this object in every send?

File: sdk/spring/spring-messaging-azure-eventhubs/src/main/java/com/azure/spring/eventhubs/core/producer/ProducerAsyncClient.java [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    }
+
+    @Override
+    public Mono<Void> send(Flux<EventData> events) {
+        return send(events, null);
+    }
+
+    @Override
+    public Mono<Void> send(Flux<EventData> events, PartitionSupplier partitionSupplier) {
+        return null;
```
Comment (stliu) [inline sdk/spring/spring-messaging-azure-eventhubs/src/main/java/com/azure/spring/eventhubs/core/producer/ProducerAsyncClient.java:35]: null?

File: sdk/spring/spring-cloud-azure-stream-binder-eventhubs/README.md [hunk 5]
```diff
@@ -120,6 +120,18 @@ Effective only if `sync` is set to true. The amount of time to wait for a respon
 
 Default: `10000`
 
+**_max-batch-in-bytes_**
```
Comment (stliu) [inline sdk/spring/spring-cloud-azure-stream-binder-eventhubs/README.md:123]: max batch **size** in bytes?

**Comment by saragluna** [comment]
Closing this PR now for no final design for the batch producer has been converged.

**Pull request closed without merging**